test/system/libpod_suite_test.go

```diff
@@ -0,0 +1,696 @@
+	}
+
+	if _, ok := p.PodmanCmdOptions[subCmd]; ok {
+		m := make(map[string]bool)
+		subArgs = p.PodmanCmdOptions[subCmd]
+		for i := 0; i < len(subArgs); i++ {
+			m[subArgs[i]] = true
+		}
+		for i := 1; i < len(args); i++ {
+			if _, ok := m[args[i]]; !ok {
+				subArgs = append(subArgs, args[i])
+			}
+		}
+	} else {
+		subArgs = args[1:]
+	}
+
+	addOptions = append(addOptions, subArgs...)
+
+	return addOptions
+}
+
+// Podman is the exec call to podman on the filesystem, uid and gid the credentials to use
+func (p *PodmanTest) PodmanAsUser(args []string, uid, gid uint32, env []string) *PodmanSession {
+	podmanOptions := p.MakeOptions(args)
+	if os.Getenv("HOOK_OPTION") != "" {
+		podmanOptions = append(podmanOptions, os.Getenv("HOOK_OPTION"))
+	}
+	if env == nil {
+		fmt.Printf("Running: %s %s\n", p.PodmanBinary, strings.Join(podmanOptions, " "))
+	} else {
+		fmt.Printf("Running: (env: %v) %s %s\n", env, p.PodmanBinary, strings.Join(podmanOptions, " "))
+	}
+	command := exec.Command(p.PodmanBinary, podmanOptions...)
+
+	if uid != 0 || gid != 0 {
+		command.SysProcAttr = &syscall.SysProcAttr{}
+		command.SysProcAttr.Credential = &syscall.Credential{Uid: uid, Gid: gid}
+	}
+	if env != nil {
+		command.Env = env
+	}
+
+	session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
+	if err != nil {
+		Fail(fmt.Sprintf("unable to run podman command: %s\n%v", strings.Join(podmanOptions, " "), err))
+	}
+	return &PodmanSession{session}
+}
+
+// Podman is the exec call to podman on the filesystem
+func (p *PodmanTest) Podman(args []string) *PodmanSession {
+	return p.PodmanAsUser(args, 0, 0, nil)
+}
+
+//WaitForContainer waits on a started container
+func WaitForContainer(p *PodmanTest) bool {
+	for i := 0; i < 10; i++ {
+		if p.NumberOfRunningContainers() == 1 {
+			return true
+		}
+		time.Sleep(1 * time.Second)
+	}
+	return false
+}
+
+// Cleanup cleans up the temporary store
+func (p *PodmanTest) Cleanup() {
+	// Remove all containers
+	stopall := p.Podman([]string{"stop", "-a", "--timeout", "0"})
+	stopall.WaitWithDefaultTimeout()
+
+	session := p.Podman([]string{"rm", "-fa"})
+	session.Wait(90)
+	// Nuke tempdir
+	if err := os.RemoveAll(p.TempDir); err != nil {
+		fmt.Printf("%q\n", err)
+	}
+
+	// Clean up the registries configuration file ENV variable set in Create
+	resetRegistriesConfigEnv()
+}
+
+// CleanupPod cleans up the temporary store
+func (p *PodmanTest) CleanupPod() {
+	// Remove all containers
+	session := p.Podman([]string{"pod", "rm", "-fa"})
+	session.Wait(90)
+	// Nuke tempdir
+	if err := os.RemoveAll(p.TempDir); err != nil {
+		fmt.Printf("%q\n", err)
+	}
+}
+
+// GrepString takes session output and behaves like grep. it returns a bool
+// if successful and an array of strings on positive matches
+func (s *PodmanSession) GrepString(term string) (bool, []string) {
+	var (
+		greps   []string
+		matches bool
+	)
+
+	for _, line := range strings.Split(s.OutputToString(), "\n") {
+		if strings.Contains(line, term) {
```

Comment:
This would be much more useful if performed a regular-expression search instead of a simple match.  Otherwise it could be confusing to use 'Grep' in the function name, since that's always associated with regular expression terms.

Reply:
The ``regexp`` package, [provides a standard implementation.](https://golang.org/pkg/regexp/#Regexp.FindAll)

Reply:
Yes. I know the package and use it in the version test. But currently this function is a little different than using a regexp, it return the whole line. If we want to use some regexp for this, we must update OutputToString() first as that function will return a one line string instead of multi-lines. 
So I'd like keep this to next pullreq for the common functions improve.